### PR TITLE
[G_70] Gayatri | Implement /v1/patients/find

### PR DESCRIPTION
**Pull request by gayatrijagtap**

### CODE REVIEW

File: src/main/java/in/projecteka/gateway/user/UserController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@RestController
+@AllArgsConstructor
+public class UserController {
+    RequestOrchestrator<PatientSearchServiceClient> patientSearchServiceClient;
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @PostMapping("/v1/patients/find")
+    public Mono<Void> findPatient(HttpEntity<String> requestEntity) {
+        Mono<Void> toBeFiredAndForgotten = patientSearchServiceClient.processRequest(requestEntity, X_CM_ID);
```
Comment (angshu) [inline src/main/java/in/projecteka/gateway/user/UserController.java]: where is this method "processRequest" defined?

File: src/main/java/in/projecteka/gateway/clients/PatientSearchServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public Mono<Void> routeRequest(Map<String, Object> request, String url) {
+        return super.routeRequest(request, url + REQUEST_ROUTE);
+    }
+
+
+    @Override
+    public Mono<Void> notifyError(ErrorResult request) {
+        Optional<YamlRegistryMapping> config = bridgeRegistry.getConfigFor(TEMP_HIU_ID, ServiceType.HIU);
```
Comment (angshu) [inline src/main/java/in/projecteka/gateway/clients/PatientSearchServiceClient.java]: why do we have a TEMP_HIU_ID defined in the constant? and why are we querying the bridgeRegistry with such a constant? Should we not be extracting from request Header?

**Review by angshu — CHANGES_REQUESTED** [review]

File: src/main/java/in/projecteka/gateway/user/UserController.java (hunk 1 above)
Comment (gayatrijagtap) [reply src/main/java/in/projecteka/gateway/user/UserController.java]: it's in RequestOrchestrator. I think, the patientSearchServiceClient should be renamed to patientSearchRequestOrchestrator.

Comment (gayatrijagtap) [reply src/main/java/in/projecteka/gateway/user/UserController.java:23]: it's in RequestOrchestrator. the patientSearchServiceClient is actually the patientSearchRequestOrchestrator. I think, I should rename it.

File: src/main/java/in/projecteka/gateway/clients/PatientSearchServiceClient.java (hunk 2 above)
Comment (gayatrijagtap) [reply src/main/java/in/projecteka/gateway/clients/PatientSearchServiceClient.java:38]: Yeah, actually. But, I am not sure, as we are using the constants only in other service clients as well.

Comment (gayatrijagtap) [reply src/main/java/in/projecteka/gateway/clients/PatientSearchServiceClient.java]: we are not passing the id from request headers

Comment (gayatrijagtap) [reply src/main/java/in/projecteka/gateway/clients/PatientSearchServiceClient.java]: done

**Review by angshu — APPROVED** [review]

**Pull request merged**